packages/sdk/src/protocols/store.ts

```diff
@@ -0,0 +1,317 @@
+import { sha256 } from "@noble/hashes/sha256";
+import { StoreCore, waku_store } from "@waku/core";
+import {
+  Cursor,
+  IDecodedMessage,
+  IDecoder,
+  IStoreSDK,
+  type Libp2p,
+  PageDirection,
+  type ProtocolCreateOptions
+} from "@waku/interfaces";
+import { ensurePubsubTopicIsConfigured, isDefined } from "@waku/utils";
+import { concat } from "@waku/utils/bytes";
+
+import { utf8ToBytes } from "../index.js";
+
+import { BaseProtocolSDK } from "./base_protocol.js";
+
+export const DefaultPageSize = 10;
+
+// const log = new Logger("sdk:store");
+
+const DEFAULT_NUM_PEERS = 1;
+
+export class StoreSDK extends BaseProtocolSDK implements IStoreSDK {
+  public readonly protocol: StoreCore;
+
+  constructor(libp2p: Libp2p, options?: ProtocolCreateOptions) {
+    // options.numPeersToUse is disregarded: https://github.com/waku-org/js-waku/issues/1685
+    super({ numPeersToUse: DEFAULT_NUM_PEERS });
+
+    this.protocol = new StoreCore(libp2p, options);
+  }
+
+  /**
+   * Do a query to a Waku Store to retrieve historical/missed messages.
+   *
+   * This is a generator, useful if you want most control on how messages
+   * are processed.
+   *
+   * The order of the messages returned by the remote Waku node SHOULD BE
+   * as follows:
+   * - within a page, messages SHOULD be ordered from oldest to most recent
+   * - pages direction depends on { @link QueryOptions.pageDirection }
+   * @throws If not able to reach a Waku Store peer to query,
+   * or if an error is encountered when processing the reply,
+   * or if two decoders with the same content topic are passed.
+   *
+   * This API only supports querying a single pubsub topic at a time.
+   * If multiple decoders are provided, they must all have the same pubsub topic.
+   * @throws If multiple decoders with different pubsub topics are provided.
+   * @throws If no decoders are provided.
+   * @throws If no decoders are found for the provided pubsub topic.
+   */
+  async *queryGenerator<T extends IDecodedMessage>(
+    decoders: IDecoder<T>[],
+    options?: waku_store.QueryOptions
+  ): AsyncGenerator<Promise<T | undefined>[]> {
+    const { pubsubTopic, contentTopics, decodersAsMap } =
+      this.validateDecodersAndPubsubTopic(decoders, options);
+
+    const queryOpts = this.constructOptions(
+      pubsubTopic,
+      contentTopics,
+      options
+    );
+
+    const peer = (
+      await this.protocol.getPeers({
+        numPeers: this.numPeers,
+        maxBootstrapPeers: 1
+      })
+    )[0];
+
+    const responseGenerator = this.protocol.queryPerPage(
+      queryOpts,
+      decodersAsMap,
+      peer
+    );
+
+    for await (const messages of responseGenerator) {
+      yield messages;
+    }
+  }
+
+  /**
+   * @deprecated Use `queryWithOrderedCallback` instead
+   **/
+  queryOrderedCallback = this.queryWithOrderedCallback;
```

Comment:
can't we remove it? anyway package is moved

Reply:
good point. this PR already introduces breaking changes, so might as well delete a deprecated function.
plus: this function has been tagged as deprecated for a lot of RCs now :)

addressed: 087afaf16ad508f0000bf20f99cc957c0e3dce15